### PR TITLE
fix: updates the httpClientDefault

### PR DESCRIPTION
**Pull request by carlossteinzup**
Updates the HttpCLientDefault on Beagle Defaults

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle-scaffold-sample/src/main/java/br/com/zup/beagle/scaffold/sample/MainActivity.kt [hunk 1]
```diff
@@ -19,17 +19,67 @@ package br.com.zup.beagle.scaffold.sample
...
+import br.com.zup.beagle.android.action.SetContext
+import br.com.zup.beagle.android.components.TabBar
+import br.com.zup.beagle.android.components.TabBarItem
+import br.com.zup.beagle.android.components.Text
+import br.com.zup.beagle.android.components.layout.Container
+import br.com.zup.beagle.android.components.layout.Screen
+import br.com.zup.beagle.android.components.page.PageView
+import br.com.zup.beagle.android.context.ContextData
+import br.com.zup.beagle.android.context.expressionOf
+import br.com.zup.beagle.android.utils.toView
```
Comment (botbeaglezup) [inline android/beagle-scaffold-sample/src/main/java/br/com/zup/beagle/scaffold/sample/MainActivity.kt:31]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle-helpers/blob/ad96d1fccb694868940a540b1572c9290df32f2c/android/beagle-scaffold-sample/src/main/java/br/com/zup/beagle/scaffold/sample/MainActivity.kt#L31"></span>The import 'br.com.zup.beagle.android.utils.toView' is unused.</td>
    </tr>
  </tbody>
</table>

File: android/beagle-scaffold-sample/src/main/java/br/com/zup/beagle/scaffold/sample/MainActivity.kt [hunk 2]
```diff
@@ -19,17 +19,67 @@ package br.com.zup.beagle.scaffold.sample
...
+import br.com.zup.beagle.scaffold.BeagleDeclarativeSample
```
Comment (botbeaglezup) [inline android/beagle-scaffold-sample/src/main/java/br/com/zup/beagle/scaffold/sample/MainActivity.kt:32]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle-helpers/blob/ad96d1fccb694868940a540b1572c9290df32f2c/android/beagle-scaffold-sample/src/main/java/br/com/zup/beagle/scaffold/sample/MainActivity.kt#L32"></span>The import 'br.com.zup.beagle.scaffold.BeagleDeclarativeSample' is unused.</td>
    </tr>
  </tbody>
</table>

File: android/beagle-defaults/build.gradle [hunk 3]
```diff
@@ -53,6 +53,7 @@ dependencies {
     implementation 'androidx.core:core-ktx:1.3.2'
```
Comment (botbeaglezup) [inline android/beagle-defaults/build.gradle:53]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle-helpers/blob/1245b6c2c01fe60088e79756c0512fb98650cc28/android/beagle-defaults/build.gradle#L53"></span>A newer version of androidx.core:core-ktx than 1.3.2 is available: 1.5.0</td>
    </tr>
  </tbody>
</table>

File: android/beagle-defaults/build.gradle [hunk 4]
```diff
@@ -53,6 +53,7 @@ dependencies {
...
     implementation 'androidx.appcompat:appcompat:1.2.0'
```
Comment (botbeaglezup) [inline android/beagle-defaults/build.gradle:54]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle-helpers/blob/1245b6c2c01fe60088e79756c0512fb98650cc28/android/beagle-defaults/build.gradle#L54"></span>A newer version of androidx.appcompat:appcompat than 1.2.0 is available: 1.3.0</td>
    </tr>
  </tbody>
</table>

File: android/beagle-defaults/build.gradle [hunk 5]
```diff
@@ -53,6 +53,7 @@ dependencies {
...
     implementation 'com.google.android.material:material:1.3.0'
+    implementation 'com.google.code.gson:gson:2.8.6'
 
     implementation "br.com.zup.beagle:android:$beagle_version"
     implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-android:1.3.9'
```
Comment (botbeaglezup) [inline android/beagle-defaults/build.gradle:59]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle-helpers/blob/1245b6c2c01fe60088e79756c0512fb98650cc28/android/beagle-defaults/build.gradle#L59"></span>A newer version of org.jetbrains.kotlinx:kotlinx-coroutines-android than 1.3.9 is available: 1.4.2</td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          6 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-helpers/blob/1245b6c2c01fe60088e79756c0512fb98650cc28/android/beagle-defaults/build.gradle#L60">android/beagle-defaults/build.gradle#L60</a> - A newer version of org.jetbrains.kotlinx:kotlinx-coroutines-test than 1.3.9 is available: 1.4.2</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-helpers/blob/1245b6c2c01fe60088e79756c0512fb98650cc28/android/beagle-defaults/build.gradle#L66">android/beagle-defaults/build.gradle#L66</a> - A newer version of io.mockk:mockk than 1.10.2 is available: 1.10.6</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-helpers/blob/1245b6c2c01fe60088e79756c0512fb98650cc28/android/beagle-defaults/build.gradle#L72">android/beagle-defaults/build.gradle#L72</a> - Avoid using + in version numbers; can lead to unpredictable and unrepeatable builds (junit:junit:4.+)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-helpers/blob/1245b6c2c01fe60088e79756c0512fb98650cc28/android/beagle-defaults/build.gradle#L74">android/beagle-defaults/build.gradle#L74</a> - A newer version of androidx.test:core than 1.4.0-alpha04 is available: 1.4.0-beta01</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-helpers/blob/1245b6c2c01fe60088e79756c0512fb98650cc28/android/beagle-defaults/build.gradle#L75">android/beagle-defaults/build.gradle#L75</a> - A newer version of androidx.test:runner than 1.4.0-alpha04 is available: 1.4.0-beta01</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-helpers/blob/1245b6c2c01fe60088e79756c0512fb98650cc28/android/beagle-defaults/build.gradle#L76">android/beagle-defaults/build.gradle#L76</a> - A newer version of androidx.test.ext:junit than 1.1.3-alpha04 is available: 1.1.3-beta01</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle-defaults/src/main/java/br/com/zup/beagle/defaults/httpclient/HttpClientDefault.kt [hunk 6]
```diff
@@ -18,17 +18,14 @@
 package br.com.zup.beagle.defaults.httpclient
 
 import br.com.zup.beagle.android.exception.BeagleApiException
-import br.com.zup.beagle.android.networking.HttpClient
-import br.com.zup.beagle.android.networking.HttpMethod
-import br.com.zup.beagle.android.networking.RequestCall
-import br.com.zup.beagle.android.networking.RequestData
-import br.com.zup.beagle.android.networking.ResponseData
+import br.com.zup.beagle.android.networking.*
```
Comment (botbeaglezup) [inline android/beagle-defaults/src/main/java/br/com/zup/beagle/defaults/httpclient/HttpClientDefault.kt:21]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle-helpers/blob/3551fb65d5dc76116429235a82f90c54db5986da/android/beagle-defaults/src/main/java/br/com/zup/beagle/defaults/httpclient/HttpClientDefault.kt#L21"></span>br.com.zup.beagle.android.networking.* is a wildcard import. Replace it with fully qualified imports.</td>
    </tr>
  </tbody>
</table>

File: android/beagle-defaults/build.gradle [hunk 7]
```diff
@@ -50,9 +50,10 @@ android {
...
     implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
-    implementation 'androidx.core:core-ktx:1.3.2'
-    implementation 'androidx.appcompat:appcompat:1.2.0'
+    implementation 'androidx.core:core-ktx:1.5.0'
+    implementation 'androidx.appcompat:appcompat:1.3.0'
     implementation 'com.google.android.material:material:1.3.0'
+    implementation 'com.google.code.gson:gson:2.8.6'
 
     implementation "br.com.zup.beagle:android:$beagle_version"
     implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-android:1.3.9'
```
Comment (botbeaglezup) [inline android/beagle-defaults/build.gradle:59]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle-helpers/blob/1de9777e918e03e97202179fdaa18877efc4982a/android/beagle-defaults/build.gradle#L59"></span>A newer version of org.jetbrains.kotlinx:kotlinx-coroutines-android than 1.3.9 is available: 1.4.2</td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          6 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-helpers/blob/1de9777e918e03e97202179fdaa18877efc4982a/android/beagle-defaults/build.gradle#L60">android/beagle-defaults/build.gradle#L60</a> - A newer version of org.jetbrains.kotlinx:kotlinx-coroutines-test than 1.3.9 is available: 1.4.2</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-helpers/blob/1de9777e918e03e97202179fdaa18877efc4982a/android/beagle-defaults/build.gradle#L66">android/beagle-defaults/build.gradle#L66</a> - A newer version of io.mockk:mockk than 1.10.2 is available: 1.10.6</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-helpers/blob/1de9777e918e03e97202179fdaa18877efc4982a/android/beagle-defaults/build.gradle#L72">android/beagle-defaults/build.gradle#L72</a> - Avoid using + in version numbers; can lead to unpredictable and unrepeatable builds (junit:junit:4.+)</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-helpers/blob/1de9777e918e03e97202179fdaa18877efc4982a/android/beagle-defaults/build.gradle#L74">android/beagle-defaults/build.gradle#L74</a> - A newer version of androidx.test:core than 1.4.0-alpha04 is available: 1.4.0-beta01</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-helpers/blob/1de9777e918e03e97202179fdaa18877efc4982a/android/beagle-defaults/build.gradle#L75">android/beagle-defaults/build.gradle#L75</a> - A newer version of androidx.test:runner than 1.4.0-alpha04 is available: 1.4.0-beta01</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-helpers/blob/1de9777e918e03e97202179fdaa18877efc4982a/android/beagle-defaults/build.gradle#L76">android/beagle-defaults/build.gradle#L76</a> - A newer version of androidx.test.ext:junit than 1.1.3-alpha04 is available: 1.1.3-beta01</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by luisoliveirazup — APPROVED** [review]

**Pull request merged**